lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4966,6 +4966,12 @@ namespace {
                     Impl.importSourceLoc(decl->getLocation()),
                     /*genericparams=*/nullptr, dc);
 
+      if (auto *GTD = dyn_cast<GenericTypeDecl>(typeDecl)) {
+        typealias->setGenericEnvironment(GTD->getGenericEnvironment());
+        if (GTD->isGeneric())
+          typealias->setGenericParams(GTD->getGenericParams()->clone(dc));
```

Comment:
Doesn't the typealias have to be the DeclContext?

Reply:
Do you mean `GenericContext`? I'm checking for `GenericTypeDecl` here because `typeDecl` already is a `TypeDecl` so not sure if it makes sense to check lower than that?

Reply:
No, sorry, I mean in the call to `clone`.

Reply:
Ah, sure, I'll change that!